OutOfSchool/OutOfSchool.WebApi.Tests/Extensions/MapSearchResultToOkOrNoContentExtensionTests.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System.Collections.Generic;
+using FluentAssertions;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using NUnit.Framework;
+using OutOfSchool.BusinessLogic.Models;
+using OutOfSchool.BusinessLogic.Models.Application;
+using OutOfSchool.WebApi.Extensions;
+
+namespace OutOfSchool.WebApi.Tests.Extensions;
+
+[TestFixture]
+public class MapSearchResultToOkOrNoContentExtensionTests
+{
+    private class TestController : ControllerBase
+    {
+    }
+
+    private TestController controller;
+
+    [SetUp]
+    public void SetUp()
+    {
+        controller = new TestController();
+    }
+
+    [Test]
+    public void MapSearchResultToOkOrNoContentExtension_SearchResultIsNotNullAndNotEmpty_ReturnOk()
+    {
+        // Arrange
+        var applications = new List<ApplicationDto>() {
+            new ApplicationDto() { RejectionMessage = "Test 1" },
+            new ApplicationDto() { RejectionMessage = "Test 2", },
+            new ApplicationDto() { RejectionMessage = "Test 3" },
+        };
+
+        var searchResult = new SearchResult<ApplicationDto>() { TotalAmount = applications.Count, Entities = applications };
+
+        // Act
+        var result = controller.MapSearchResultToOkOrNoContent(searchResult);
```

Comment:
(Test mocks) As mentioned previously `controller` here is not the unit we are testing (not the system-under-test, or SUT), it's just a dependency for the real SUT - extension method. As all dependencies it could be effectively mocked using Mock library. Please rewrite

Reply:
Fixed.